Scripts/SAPSetupAssistantValidationScripts/SetupAssistantSAPValidateNetworkConnection.ps1

```diff
@@ -0,0 +1,29 @@
+param (
+   [string]$hostName,
+   [string]$port
+)
+
+$job = Start-Job -ScriptBlock {
+    param($hostName, $port)
+    Test-NetConnection -ComputerName $hostName -Port $port
+} -ArgumentList $hostName, $port
+
+for ($i = 0; $i -lt 5; $i++) {
+    Start-Sleep -Seconds 1
+}
+
+if (Wait-Job -Job $job -Timeout 5) {
+    $result = Receive-Job -Job $job
+
+    if ($result.TcpTestSucceeded) {
+        Write-Host -NoNewLine "[{'step': 'CheckNetworkConnection', 'status': 'Success', 'message': 'Successfully connected to %hostName% over port %port%.'}]"
+    }
+    else {
+         Write-Host -NoNewLine "[{'step': 'CheckNetworkConnection', 'status': 'Error', 'message': 'Failed to connect to %hostName% over port %port%.'}]"
+    }
+}
+else {
+    Write-Host -NoNewLine "[{'step': 'CheckNetworkConnection', 'status': 'Error', 'message': 'The connection to %hostName% over port %port% timed out after 5 seconds.'}]"
```

Comment:
The output appears to be json, but not valid JSON. might try using a variable like `$output` and then passing that through `ConvertTo-Json` like so:

```
$output | ConvertTo-Json -Compress
```

This will ensure valid JSON is used. 

You can assign the information as an object at each condition like so:

```
$output = @{
    step = "CheckNetworkConnection"
    status = "Success"
    message = "Successfully connected to $HostName over port $Port."
}
```

Reply:
Currently we are only showing the result to the user in the terminal, the result is not being anywhere else. So, I think we can leave it as is for now. Let me know your thoughts.